server/routes/punishmentComment/punishmentCommentDeletePage.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable max-classes-per-file */
+import { Request, Response } from 'express'
+import url from 'url'
+import { FormError } from '../../@types/template'
+import UserService from '../../services/userService'
+import { hasAnyRole } from '../../utils/utils'
+import adjudicationUrls from '../../utils/urlGenerator'
+import PunishmentsService from '../../services/punishmentsService'
+
+type PageData = {
+  error?: FormError
+  punishmentComment?: string
+}
+
+export default class ConfirmDeletionPage {
+  constructor(private readonly userService: UserService, private readonly punishmentsService: PunishmentsService) {}
+
+  private renderView = async (req: Request, res: Response, pageData: PageData): Promise<void> => {
+    const { error, punishmentComment } = pageData
+
+    return res.render(`pages/punishmentCommentConfirmDeletion.njk`, {
+      errors: error ? [error] : [],
+      punishmentComment,
+    })
+  }
+
+  view = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const adjudicationNumber = Number(req.params.adjudicationNumber)
+    const punishmentComments = await this.punishmentsService.getPunishmentCommentsFromServer(adjudicationNumber, user)
+
+    const id = Number(req.params.id)
+    const punishmentComment = punishmentComments.find(comment => comment.id === id)
+    if (!punishmentComment) {
+      return res.render('pages/notFound.njk', {
+        url: adjudicationUrls.punishmentsAndDamages.urls.review(adjudicationNumber),
+      })
+    }
+
+    return this.renderView(req, res, { punishmentComment: punishmentComment.comment })
+  }
+
+  submit = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const userRoles = await this.userService.getUserRoles(res.locals.user.token)
+    const adjudicationNumber = Number(req.params.adjudicationNumber)
+    const id = Number(req.params.id)
+    const { removeComment } = req.body
+
+    if (removeComment === undefined) {
+      const error = { text: 'Please select required option' } as FormError
```

Comment:
Errors need to take the form e.g.
  MISSING_REASON: {
    href: '#adjournReason',
    text: 'Enter the reason for the adjournment',
  },
  
  so that when they render on the page, it highlights the field that the user needs to focus on. So in this case, you need to include a href which is the id of the field in the nunjucks file, like `#punishmentComment`
/

Reply:
fixed.